actionpack/lib/action_dispatch/system_testing/driver.rb

```diff
@@ -3,17 +3,21 @@
 module ActionDispatch
   module SystemTesting
     class Driver # :nodoc:
-      def initialize(name, **options, &capabilities)
-        @name = name
+      attr_reader :name
+
+      def initialize(driver_type, **options, &capabilities)
+        @driver_type = driver_type
         @browser = Browser.new(options[:using])
         @screen_size = options[:screen_size]
         @options = options[:options] || {}
         @capabilities = capabilities
 
-        if name == :selenium
+        if driver_type == :selenium
           require "selenium/webdriver"
           @browser.preload
         end
+
+        @name = options[:name] || [driver_type, options[:using], object_id].compact.map(&:to_s).join("_").to_sym
```

Comment:
`[driver_type, options[:using], object_id].compact.map(&:to_s).join("_").to_sym` feels needlessly verbose. I don't understand why we wouldn't use only `driver_type` if name is not supplied.

Reply:
The PR as written basically just takes the stance that the Capybara driver name should be guaranteed to be unique if left unspecified because that's the least surprising behavior. Users using `driven_by` might not necessarily understand that the Capybara driver name is derived entirely from `driver_type` and that they would need to do additional work (specifying the `name` explicitly) in order to avoid conflicts.

The downside, of course, is that this would break existing test code that has a dependency on the specific Capybara driver name.

I'm fine with making the default Capybara driver name just the `driver_type` (essentially mimicking the current behavior), as long as it's clear that doing so doesn't actually address #42502 in a particularly satisfying way: it puts the burden on the user to recognize that they've got multiple drivers with the same `driver_type` (and thus, the same Capybara driver name).

Reply:
One option that might be nice is use the `driver_type` as the default driver name as you suggest, but do it in a way that if the user registers multiple Capybara drivers with the same name, we raise an exception.

That way we can keep existing driver names in tact, but we also don't surprise users if there's a conflict.

Reply:
> The PR as written basically just takes the stance that the Capybara driver name should be guaranteed to be unique if left unspecified because that's the least surprising behavior

I don't agree with this. I wouldn't expect the driver name to be unique. Maybe that's because I'm the author of this feature, but if I did not set a custom name I would be very surprised if Rails returned a combined name with an object id in it.

> The downside, of course, is that this would break existing test code that has a dependency on the specific Capybara driver name.

We definitely can't merge this as is without a deprecation. We should avoid changing existing behavior without a deprecation cycle.

> it puts the burden on the user to recognize that they've got multiple drivers with the same driver_type (and thus, the same Capybara driver name)

But that's how Capybara works. I think if someone is adding multiple drivers with the same type then application authors need to be aware of how to customize that.

Reply:
How do you feel about raising an exception if the same Capybara driver name is registered multiple times?